### PR TITLE
Experimental/sf

### PR DESCRIPTION
**Pull request by ndquocphong**

### CODE REVIEW

File: Tests/Web/BlockRecommendControllerTest.php [hunk 1]
```diff
@@ -38,11 +48,13 @@ public function testRecommendBlock()
     {
         $crawler = $this->client->request(
             'GET',
-            $this->app->url('block_recommend_product_block')
+            $this->generateUrl('block_recommend_product_block')
```
Comment (ndquocphong) [inline Tests/Web/BlockRecommendControllerTest.php:51]: Unit test failed due to exception 
```
Twig_Error_Loader: Unable to find template "Block/recommend_product_block.twig" 
```

File: Controller/RecommendController.php [hunk 2]
```diff
@@ -10,32 +10,59 @@
...
+     * @Route("/%eccube_admin_route%/plugin/recommend/", name="plugin_recommend_list")
      */
-    public function index(Application $app, Request $request)
+    public function index(Request $request)
     {
-        $pagination = $app['eccube.plugin.recommend.repository.recommend_product']->getRecommendList();
+        $pagination = $this->recommendProductRepository->getRecommendList();
 
-        return $app->render('Recommend/Resource/template/admin/index.twig', array(
+        return $this->render('Recommend/Resource/template/admin/index.twig', array(
```
Comment (ndquocphong) [inline Controller/RecommendController.php]: Try use 
```
@Template("@Recommend/admin/index.twig")
```
to consistent with Controller/Block/RecommendController.php

File: Tests/Repository/RecommendRepositoryTest.php [hunk 3]
```diff
@@ -11,35 +11,59 @@
 namespace Plugin\Recommend\Tests\Repository;
 
 use Eccube\Common\Constant;
+use Eccube\Entity\Product;
```
Comment (ndquocphong) [inline Tests/Repository/RecommendRepositoryTest.php]: Please remove unused code 
```
use Eccube\Entity\Product;
```

File: Tests/Web/BlockRecommendControllerTest.php (hunk 1 above)
Comment (lqdung-lockon) [reply Tests/Web/BlockRecommendControllerTest.php:51]: please re-enable again

File: Nav.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+     * {@inheritdoc}
+     *
+     * @return array
+     */
+    public static function getNav()
+    {
+        return [
+            'content' => [
+                'id' => 'plugin_recommend',
+                'name' => 'おすすめ管理',
```
Comment (ndquocphong) [inline Nav.php]: Should use translate

**Comment by hoand-vn** [comment]
fixed comment

**Pull request closed without merging**